From 27ded21fea2a6f1fc161a91744014cce711ad1b3 Mon Sep 17 00:00:00 2001
From: Alistair Francis <Alistair.Francis@wdc.com>
Date: Mon, 10 Dec 2018 17:34:29 +0000
Subject: [PATCH Libgpg-error] syscfg: Add a riscv32 architecture

* src/syscfg/lock-obj-pub.riscv32-unknown-linux-gnu.h: New.
* src/Makefile.am (lock_obj_pub): Add it.

Signed-off-by: Alistair Francis <alistair.francis@wdc.com>
Signed-off-by: pino-kim <sungwon.pino@gmail.com>
Upstream-Status: Accepted [1.34]
---
 src/Makefile.am                               |  1 +
 .../lock-obj-pub.riscv32-unknown-linux-gnu.h  | 23 +++++++++++++++++++
 2 files changed, 24 insertions(+)
 create mode 100644 src/syscfg/lock-obj-pub.riscv32-unknown-linux-gnu.h

diff --git a/src/Makefile.am b/src/Makefile.am
index 8e6683f..ce1b882 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -67,6 +67,7 @@ lock_obj_pub = \
 	syscfg/lock-obj-pub.powerpc64le-unknown-linux-gnu.h \
 	syscfg/lock-obj-pub.powerpc-unknown-linux-gnuspe.h  \
 	syscfg/lock-obj-pub.riscv64-unknown-linux-gnu.h     \
+	syscfg/lock-obj-pub.riscv32-unknown-linux-gnu.h     \
         syscfg/lock-obj-pub.s390x-unknown-linux-gnu.h       \
         syscfg/lock-obj-pub.sh3-unknown-linux-gnu.h         \
         syscfg/lock-obj-pub.sh4-unknown-linux-gnu.h         \
diff --git a/src/syscfg/lock-obj-pub.riscv32-unknown-linux-gnu.h b/src/syscfg/lock-obj-pub.riscv32-unknown-linux-gnu.h
new file mode 100644
index 0000000..b5bdaf5
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.riscv32-unknown-linux-gnu.h
@@ -0,0 +1,23 @@
+## lock-obj-pub.riscv32-unknown-linux-gnu.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
-- 
2.19.1

